lib/ansible/modules/cloud/vmware/vmware_guest_register_operation.py

```diff
@@ -0,0 +1,263 @@
+        if(not(datacenter)):
+            self.module.fail_json(msg="Cannot find the specified Datacenter: %s" % self.datacenter)
+
+        dcpath = compile_folder_path_for_object(datacenter)
+        if(not(dcpath.endswith("/"))):
+            dcpath += "/"
+
+        if(self.folder in [None, "", "/"]):
+            self.module.fail_json(msg="Please specify folder path other than blank or '/'")
+        elif(self.folder.startswith("/vm")):
+            fullpath = "%s%s%s" % (dcpath, self.datacenter, self.folder)
+        else:
+            fullpath = "%s%s" % (dcpath, self.folder)
+
+        folder_obj = self.content.searchIndex.FindByInventoryPath(inventoryPath="%s" % fullpath)
+        if(not(folder_obj)):
+            details = {
+                'datacenter': datacenter.name,
+                'datacenter_path': dcpath,
+                'folder': self.folder,
+                'full_search_path': fullpath,
+            }
+            self.module.fail_json(msg="No folder %s matched in the search path : %s" % (self.folder, fullpath),
+                                  details=details)
+
+        if(self.state == "present"):
+            if(find_vm_by_name(self.content, self.name, folder=folder_obj)):
+                self.module.exit_json(**result)
+            else:
+                if(self.esxi_hostname):
+                    host_obj = self.find_hostsystem_by_name(self.esxi_hostname)
+                    if(not(host_obj)):
+                        self.module.fail_json(msg="Cannot find the specified ESXi host: %s" % self.esxi_hostname)
+                else:
+                    host_obj = None
+
+                if(self.cluster_name):
+                    cluster_obj = find_cluster_by_name(self.content, self.cluster_name, datacenter)
+                    if(not(cluster_obj)):
+                        self.module.fail_json(msg="Cannot find the specified cluster name: %s" % self.cluster_name)
+
+                    resource_pool_obj = cluster_obj.resourcePool
+                elif(self.resource_pool):
+                    resource_pool_obj = find_resource_pool_by_name(self.content, self.resource_pool)
+                    if(not(resource_pool_obj)):
+                        self.module.fail_json(msg="Cannot find the specified resource pool: %s" % self.resource_pool)
+                else:
+                    resource_pool_obj = host_obj.parent.resourcePool
+
+                task = folder_obj.RegisterVM_Task(path=self.path, name=self.name, asTemplate=self.template,
+                                                  pool=resource_pool_obj, host=host_obj)
+
+                try:
+                    wait_for_task(task)
+                except Exception as task_e:
+                    self.module.fail_json(msg=to_native(task_e))
+
+                result.update(changed=True)
+                self.module.exit_json(**result)
+
+        else:
+            vm_obj = find_vm_by_name(self.content, self.name, folder=folder_obj)
```

Comment:
Use `get_vm` API

Reply:
Modified to use get_vm from find_vm_by_name.